### PR TITLE
minor improvements to wix.yml workflow

### PR DESCRIPTION
**Pull request by gdams**
Some minor refactoring. Removes all deprecation warnings we're currently seeing in the workflow runs

### CODE REVIEW

**Review by steelhead31 — APPROVED** [review]
LGTM.

File: .github/workflows/wix.yml [hunk 1]
```diff
@@ -54,13 +52,30 @@ jobs:
     steps:
       - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
 
-      - name: Install dependencies
+      - name: Download IcedTea-Web
         run: |
-          choco install --no-progress wget
+          Invoke-WebRequest -Uri "https://github.com/AdoptOpenJDK/IcedTea-Web/releases/download/${{ matrix.ICEDTEA_WEB_VERSION }}/${{ matrix.ICEDTEA_WEB_VERSION }}.win.bin.zip" `
```
Comment (karianna) [inline .github/workflows/wix.yml:57]: I think the default further above should be upgraded to 1.8.8 (same with the Java version up above, move to latest).

File: .github/workflows/wix.yml [hunk 2]
```diff
@@ -25,7 +25,6 @@ jobs:
           SUB_TAG: 8u362b09
           JVM: hotspot
         - jdk: 11
-          ICEDTEA_WEB_VERSION: ""
```
Comment (karianna) [inline .github/workflows/wix.yml:28]: Would we not set this to 1.8.8?

File: .github/workflows/wix.yml [hunk 3]
```diff
@@ -69,38 +84,24 @@ jobs:
...
+          PRODUCT_MAINTENANCE_VERSION: ${{ matrix.PRODUCT_MAINTENANCE_VERSION }}
+          PRODUCT_PATCH_VERSION: ${{ matrix.PRODUCT_PATCH_VERSION }}
+          PRODUCT_BUILD_NUMBER: ${{ matrix.PRODUCT_BUILD_NUMBER }}
+          MSI_PRODUCT_VERSION: ${{ matrix.MSI_PRODUCT_VERSION }}
+          ARCH: ${{ matrix.ARCH }}
+          JVM: ${{ matrix.JVM }}
         shell: cmd
 
-      - uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+      - uses: actions/upload-artifact@c7d193f32edcb7bfad88892161225aeda64e9392 # v4.0.0
```
Comment (karianna) [inline .github/workflows/wix.yml:104]: Is that the latest version?

File: .github/workflows/wix.yml (hunk 2 above)
Comment (gdams) [reply .github/workflows/wix.yml:28]: ITW only works with Java 8 so we skip on 11 and 17

File: .github/workflows/wix.yml (hunk 1 above)
Comment (gdams) [reply .github/workflows/wix.yml:57]: Yeah I didn't want to over complicate this PR, I'll bump it as part of Josh's larger Wix upgrade

File: .github/workflows/wix.yml (hunk 3 above)
Comment (gdams) [reply .github/workflows/wix.yml:104]: it's to bring it up to date with the others. Dependabot will then do the next upgrade

**Review by karianna — APPROVED** [review]

File: .github/workflows/wix.yml [hunk 4]
```diff
@@ -54,13 +52,30 @@ jobs:
...
+          Remove-Item icedtea-web-image\share\doc -Recurse
+        working-directory: wix/SourceDir
+        if: ${{ matrix.ICEDTEA_WEB_VERSION }}
 
-      - name: Setup environment variables
-        uses: allenevans/set-env@0f3306034af0ea21dd28983b6e7c1614ee317739 # v3.0.0
-        with:
+      - name: Download Prebuilt JDK/JRE
+        run: |
+          Invoke-WebRequest -Uri "https://github.com/adoptium/temurin${{ matrix.PRODUCT_MAJOR_VERSION }}-binaries/releases/download/${{ matrix.TAG }}/OpenJDK${{ matrix.PRODUCT_MAJOR_VERSION }}U-jdk_${{ matrix.ARCH }}_windows_${{ matrix.JVM }}_${{ matrix.SUB_TAG }}.zip" `
```
Comment (tellison) [inline .github/workflows/wix.yml:67]: Can't we use the official Adoptium API for this download rather than reaching into the GitHub store?

Comment (karianna) [reply .github/workflows/wix.yml:67]: Argh missed that in my review, @gdams we should fix this.

**Pull request merged**